src/page-multi-facility/ScenarioLibraryModal.tsx

```diff
@@ -112,13 +112,56 @@ const IconRecidviz = styled.img`
   margin: auto;
 `;
 
+const DeleteModalContents = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  font-weight: normal;
+  justify-content: flex-start;
+  margin-top: 30px;
+`;
+
+const ModalText = styled.div`
+  font-size: 13px;
+  margin-bottom: 25px;
+`;
+
+const ModalButtons = styled.div`
+  /* display: flex;
+  flex-direction: column; */
+`;
+
+const ModalButton = styled(StyledButton)`
+  font-size: 14px;
+  font-weight: normal;
+`;
+
+const DeleteButton = styled(ModalButton)`
+  background: ${Colors.darkRed};
+  color: ${Colors.white};
+  margin-right: 15px;
+`;
+
+const CancelButton = styled(ModalButton)`
+  background: transparent;
+  border: 1px solid ${Colors.forest};
+  color: ${Colors.forest};
+`;
+
+type Props = Pick<ModalProps, "trigger">;
+
 const ScenarioLibraryModal: React.FC<Props> = ({ trigger }) => {
   const [modalOpen, setModalOpen] = useState(false);
-  const [, dispatchScenarioUpdate] = useScenario();
+  const [currentScenario, dispatchScenarioUpdate] = useScenario();
   const [scenarios, setScenarios] = useState({
     data: [] as Scenario[],
     loading: true,
   });
+  const [baselineScenario, setBaselineScenario] = useState<Scenario | null>();
+  const [scenarioIdPendingDeletion, setScenarioIdPendingDeletion] = useState<
+    string | null
+  >();
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
```

Comment:
This component is a lot more stateful that I was hoping for it to be, but hopefully it's not too difficult to follow along with what's going on.